services/telemetry/telemetry_test.go

```diff
@@ -17,17 +17,115 @@
 package telemetry
 
 import (
+	"context"
+	"encoding/hex"
+	"os"
 	"testing"
+	"time"
 
+	"github.com/golang/protobuf/proto"
+	"github.com/google/uuid"
+	pmmv1 "github.com/percona-platform/saas/gen/telemetry/events/pmm"
+	reporterv1 "github.com/percona-platform/saas/gen/telemetry/reporter"
 	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
 )
 
-func TestMakePayload(t *testing.T) {
-	s := NewService(nil, "1.3.1")
-	expected := "ECAB81E4C47D456CA9EC20AEBF91AB44;OS;\nECAB81E4C47D456CA9EC20AEBF91AB44;PMM;1.3.1\n"
-	assert.Equal(t, expected, string(s.makePayload("ECAB81E4C47D456CA9EC20AEBF91AB44"))) // \n are important
+const devTelemetryHost = "callhome-staging.percona.com:443"
+
+//nolint:lll
+func TestMakeV1Payload(t *testing.T) {
+	s := NewService(nil, "")
+
+	type param struct {
+		os      string
+		uuid    string
+		version string
+	}
+
+	for expected, p := range map[string]param{
+		"ACAB81E4C47D456CA9EC20AEBF91AB44;OS;linux\nACAB81E4C47D456CA9EC20AEBF91AB44;PMM;1.3.1\n": {os: "linux", uuid: "ACAB81E4C47D456CA9EC20AEBF91AB44", version: "1.3.1"},
+		"BCAB81E4C47D456CA9EC20AEBF91AB44;OS;ovf\nBCAB81E4C47D456CA9EC20AEBF91AB44;PMM;1.3.2\n":   {os: "ovf", uuid: "BCAB81E4C47D456CA9EC20AEBF91AB44", version: "1.3.2"},
+		"CCAB81E4C47D456CA9EC20AEBF91AB44;OS;ami\nCCAB81E4C47D456CA9EC20AEBF91AB44;PMM;1.3.3\n":   {os: "ami", uuid: "CCAB81E4C47D456CA9EC20AEBF91AB44", version: "1.3.3"},
```

Comment:
let's set version 2.x to one of them to make it clear that test not about PMM V1

Reply:
Done